features/step_definitions/helper_services.rb

```diff
@@ -153,19 +153,48 @@
   step %Q/evaluation of `pod` is stored in the :proxy_pod clipboard/
 end
 
-Given /^I have LDAP service in my project$/ do
-    ###
-    # The original idea is trying to put ldap server in the openshift to make this flexy.
-    # Since we run the scenario in jenkins slave which is not in sdn, then two choices come to me:
-    # 1, Create a route for the ldapserver pod, but blocked by this us https://trello.com/c/9TXvMeS2 is done.
-    # 2, Port forward the ldap server pod to the jenkins slave.
-    # So take the second one since this one can be implemented currently
-    ###
+###
+# The original idea is trying to put ldap server in the openshift to make this flexy.
+# Since we run the scenario in jenkins slave which is not in sdn, then two choices come to me:
+# 1, Create a route for the ldapserver pod, but blocked by this us https://trello.com/c/9TXvMeS2 is done.
+# 2, Port forward the ldap server pod to the jenkins slave.
+# So take the second one since this one can be implemented currently
+###
+Given /^I have LDAP service in my project with clusterversion "([^"]*)"$/ do |clusterversion|
+  if clusterversion == '4.5' || '4.6'
+    step 'I obtain test data file "groups/replicacontroller.yaml"'
+    step %Q/I run the :create admin command with:/, table(%{
+            | f         | replicacontroller.yaml |
+            | namespace | #{project.name} |
+          })
     step %Q/I run the :run client command with:/, table(%{
       | name  |ldapserver                             |
       | image |openshift/openldap-2441-centos7:latest |
       })
     step %Q/the step should succeed/
+    step %Q/the pod named "ldapserver" becomes ready/
+
+    cb.ldap_pod = BushSlicer::Pod.get_labeled(["run", "ldapserver"], user: user, project: project).first
+    cb.ldap_pod_name = cb.ldap_pod.name
+    cache_pods cb.ldap_pod
```

Comment:
i.e. you can skip creating a RC above, just do here:
```
  @result = pod.wait_till_ready(user, 300)
  raise "git service did not become ready" unless @result[:success]
```
Then you can have same code path for pre- and post- version 4.6.

Reply:
Added `    step %Q/a pod becomes ready with labels:/, table(%{
      | run=ldapserver |
      })
` and it worked for me.